radio/src/gui/common/stdlcd/radio_sdmanager.cpp

```diff
@@ -105,13 +105,14 @@ void onSdManagerMenu(const char * result)
     memcpy(reusableBuffer.sdmanager.originalName, line, sizeof(reusableBuffer.sdmanager.originalName));
     char * ext = getFileExtension(line, SD_SCREEN_FILE_LENGTH+1);
     if (ext) {
+      size_t len = strlen(ext);
```

Comment:
I think you could add a new function ```getFilenameInfo``` or something like this which does what you are describing above and the ```getFileExtension``` function could call it (this function is really simple and called in many locations)